File: src/packets/PacketS2CZoneMaxVolume.js
const PacketS2C = require("./PacketS2C");

/**
 * Server -> Client
 * ID = 0x64
 * Zone Max Volume
 * Informs client of a zone's max volume
 * Data:
 *      (Unsigned Char) Controller ID
 *      (Unsigned Char) Zone ID
 *      (Unsigned Char) Volume
 */
class PacketS2CZoneMaxVolume extends PacketS2C {
    constructor(ctrlrID, zoneID, maxVolume) {
        super();
        this._buffer.writeUInt8(ctrllrID);
        this._buffer.writeUInt8(zoneID);
        this._buffer.writeUInt8(maxVolume);
    }

    getID() {
        return 0x64;
    }
}

module.exports = PacketS2CZoneMaxVolume;

File: src/RNetPi.Core/Packets/PacketS2CZoneMaxVolume.cs
namespace RNetPi.Core.Packets;

/// <summary>
/// Server -> Client
/// ID = 0x64
/// Zone Max Volume
/// Sends a zone's current max volume
/// Data:
///     (Unsigned Char) Controller ID
///     (Unsigned Char) Zone ID
///     (Unsigned Char) Max Volume
/// </summary>
public class PacketS2CZoneMaxVolume : PacketS2C
{
    public const byte ID = 0x64;

    public PacketS2CZoneMaxVolume(byte controllerID, byte zoneID, byte maxVolume)
    {
        Writer.Write(controllerID);
        Writer.Write(zoneID);
        Writer.Write(maxVolume);
    }

    public override byte GetID() => ID;
}